### ISSUE TITLE
Expected fix for LeaseID missing not present in v0.12.0

### ISSUE THREAD

**Issue by eccles** [Azure/azure-storage-blob-go#236]
### Which version of the SDK was used?
0.12.0

### Which platform are you using? (ex: Windows, Linux, Debian)

Linux

### What problem was encountered?

Incorrect version tagged upstream - expected fix #230 not implemented

### How can we reproduce the problem in the simplest way?

go get v0.12.0 and inspect chunkwriting.go - expecetd fix from #230 not present in line 187

### Have you found a mitigation/solution?
No

### Some explanation ####
Thanks for the release - however the tagging has gone awry - I change my go.mod file to:

github.com/Azure/azure-storage-blob-go v0.12.0

and inspection of my go cache:

vi /opt/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.12.0/azblob/chunkwriting.go

shows the change has not been applied.

187 _, err := c.to.StageBlock(c.ctx, chunk.id, bytes.NewReader(chunk.buffer), **LeaseAccessConditions{}**, nil, c.o.ClientProvidedKeyOptions)

If I pull repo directly using git clone and inspect that all looks good:

paul@paul-Precision-7530:~/git/azure-storage-blob-go$ git log
commit 6df5d9a (HEAD -> master, tag: v0.12.0, origin/master, origin/HEAD)
Merge: 456ab47 559b75b
Author: Mohit Sharma 65536214+mohsha-msft@users.noreply.github.com
Date: Wed Dec 9 17:40:48 2020 +0530

Merge pull request #235 from Azure/dev

Release v0.12.0
commit 559b75b (origin/dev)
Author: Mohit Sharma 65536214+mohsha-msft@users.noreply.github.com
Date: Wed Dec 9 12:49:08 2020 +0530

v0.12.0 Release change logs (#231)

* v0.12.0 Release ChangeLogs & BreakingChanges

* Updated changelogs

* Added mmf changes in ChangeLog.md
commit 0c35e97
Author: Mohit Sharma 65536214+mohsha-msft@users.noreply.github.com
Date: Fri Nov 20 23:41:19 2020 +0530

**Fixed #230 (#232)**
and the fix is in:

186 _, err := c.to.StageBlock(c.ctx, chunk.id, bytes.NewReader(chunk.buffer), **c.o.AccessConditions.LeaseAccessConditions**, nil, c.o.ClientProvidedKeyOptions)

**Comment by eccles** [comment]
Speculation: the merge for the addition of c.o.ClientProvidedKeyOptions overwrote the change for c.o.AccessConditions.LeaseAccessConditions

**Comment by mohsha-msft** [comment]
Hey @eccles ,

Thanks for reaching out!
I noticed the changes from #232 were overwritten by #235 in MMF changes.  I apologize for error on my side. Let me just fix it again and then you can sync the changes from dev branch by running `go get github.com/Azure/azure-storage-blob-go@dev`

**Comment by eccles** [comment]
Tried it but still no change

**Comment by mohsha-msft** [comment]
hey @eccles ,

The changes which you mentioned has already been merged. Can you explain what issue you are facing now?

**Comment by eccles** [comment]
To summaru=ise the state of the repo looks correct but when I do a go get I get a versuion without the #230 change which I need.
I have no explanation of why this is so 

I have tried @dev which gives me the v0.10.1-0.20201209071908-559b75bbc31d which is no good.
I have tried @6df5d9af221d67aa1173ad090 which has the same problem..

**Comment by eccles** [comment]
go list -m -json github.com/Azure/azure-storage-blob-go@v0.12.0
{
	"Path": "github.com/Azure/azure-storage-blob-go",
	"Version": "v0.12.0",
	"Time": "**2020-11-21T18:08:49**Z",
	"Dir": "/opt/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.12.0",
	"GoMod": "/opt/go/pkg/mod/cache/download/github.com/!azure/azure-storage-blob-go/@v/v0.12.0.mod",
	"GoVersion": "1.15"
}

The time looks a little suspicious...

**Comment by h-no** [comment]
I was curious and had a look myself.

I checkout out the repo.
```
❯ git show
commit 6df5d9af221d67aa1173ad09073ef95c872beda0 (HEAD -> master, tag: v0.12.0, origin/master, origin/HEAD)
Merge: 456ab47 559b75b
Author: Mohit Sharma <65536214+mohsha-msft@users.noreply.github.com>
Date:   Wed Dec 9 17:40:48 2020 +0530

    Merge pull request #235 from Azure/dev

    Release v0.12.0
```
Then diffed the entire tree using Meld with my go mod cache.
Ended up with the following diff:
```
❯ diff ~/.go/pkg/mod/github.com/\!azure/azure-storage-blob-go@v0.12.0/azblob/chunkwriting.go /github.com/Azure/azure-storage-blob-go/azblob/chunkwriting.go
186,187c186
<
< 	_, err := c.to.StageBlock(c.ctx, chunk.id, bytes.NewReader(chunk.buffer), LeaseAccessConditions{}, nil, c.o.ClientProvidedKeyOptions)
---
> 	_, err := c.to.StageBlock(c.ctx, chunk.id, bytes.NewReader(chunk.buffer), c.o.AccessConditions.LeaseAccessConditions, nil, c.o.ClientProvidedKeyOptions)
```

So the bytes that gets downloaded by go is not what's in the repo.

Could this be the reason why I got the checksum mismatch in this issue that you closed: https://github.com/Azure/azure-storage-blob-go/issues/237

**Comment by h-no** [comment]
So to summarize, @eccles I experience the same as you.

**Comment by bcmills** [comment]
It looks like the `v0.12.0` tag was moved at some point:
```
$ GOPROXY=direct go get -d github.com/Azure/azure-storage-blob-go@6df5d9af221d67aa1173ad09073ef95c87
2beda0
go: downloading github.com/Azure/azure-storage-blob-go v0.12.0
go get github.com/Azure/azure-storage-blob-go@6df5d9af221d67aa1173ad09073ef95c872beda0: github.com/Azure/azure-storage-blob-go@v0.12.0: verifying module: checksum mismatch
        downloaded: h1:LEu0WO1uxAkeJFuNTZR+xxUpIoRUwOXgA2ipS798Ycc=
        sum.golang.org: h1:7bFXA1QB+lOK2/ASWHhp6/vnxjaeeZq6t8w1Jyp0Iaw=
SECURITY ERROR
This download does NOT match the one reported by the checksum server.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.
For more information, see 'go help module-auth'.
```

As a workaround, you can instead refer to that commit by the [pseudo-version](https://golang.org/ref/mod#pseudo-versions) `v0.11.1-0.20201209121048-6df5d9af221d`, which is what it would have been called prior to the erroneous tag being applied:
```
$ go get -d github.com/Azure/azure-storage-blob-go@v0.11.1-0.20201209121048-6df5d9af221d
go: downloading github.com/Azure/azure-storage-blob-go v0.11.1-0.20201209121048-6df5d9af221d

$
```

**Comment by bcmills** [comment]
Probably the best fix going forward would be to `retract` `v0.12.0` (see https://github.com/golang/go/issues/24031#issuecomment-597263309) and tag a new `v0.12.1` at the desired commit.

**Comment by eccles** [comment]
@bcmills thanx - @mohsha-msft ?

**Comment by vangent** [comment]
@mohsha-msft any chance for a 12.1 tag soon?

**Comment by eccles** [comment]
This issue has become a matter of urgency for us - any chance of a 0.12.1 release which is properly tagged?

**Comment by eccles** [comment]
I can confirm that using v0.11.1-0.20201209121048-6df5d9af221d as suggested by @bcmills does apparently work.

**Comment by ondrejbudai** [comment]
The workaround works for me too. I would love to see 0.12.1 released though.

**Comment by zezha-msft** [comment]
That's quite a strange issue, we apologize for the inconvenience.

A new release just went out: https://github.com/Azure/azure-storage-blob-go/releases/tag/v0.13.0

Please let us know if you are still encountering problems.

**Comment by eccles** [comment]
HI - I can confirm that this is fixed by v0.13.0
